src/notification/BaseSubscriptionHandler.ts

```diff
@@ -0,0 +1,28 @@
+import type { Readable } from 'stream';
+import type { ResourceIdentifier } from '../http/representation/ResourceIdentifier';
+import type { Guarded } from '../util/GuardedStream';
+import { AS } from '../util/Vocabularies';
+import type { Subscription } from './Subscription';
+import type { SubscriptionHandler } from './SubscriptionHandler';
+
+export abstract class BaseSubscriptionHandler<TSubscription extends Subscription>
+implements SubscriptionHandler<TSubscription> {
+  public abstract getResponseData(subscription: TSubscription): Guarded<Readable> | undefined;
+  public abstract getType(): string;
+  public abstract subscribe(request: any): TSubscription;
+  public abstract onResourceCreated(resource: ResourceIdentifier, subscription: TSubscription): void;
+  public abstract onResourceUpdated(resource: ResourceIdentifier, subscription: TSubscription): void;
+  public abstract onResourceDeleted(resource: ResourceIdentifier, subscription: TSubscription): void;
+  public async onChange(
+    resource: ResourceIdentifier,
+    activity: string,
```

Comment:
I tried typing it differently but I couldn't get it to work.

Reply:
I don't think this is possible with how the vocabularies are currently typed. It might be possible by updating `Vocabularies.ts` using [template literal types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) as those didn't exist yet when that part was initially created, but might require some fiddling.

Reply:
I gave it a brief attempt but didn't succeed 😅 Seems like I'll have to do read up on the type system of typescript a bit.

Reply:
I enjoy playing around with the typings so you caused me to make https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1422 😉

Reply:
I will make the required changes once #1422 is merged